enblend-enfuse: build error on staging-next

### Describe the bug

Build error on current staging-next ( #148396 )

### Steps To Reproduce
Checkout current staging-next and attempt to build (via a dependent like digikam for example)

### Additional context

```
include/boost/config/compiler/gcc.hpp:323: note: this is the location of the previous definition
minimizer.cc: In constructor 'Minimizer::Minimizer(size_t)':
minimizer.cc:29:53: error: 'numeric_limits' is not a member of 'std'
   29 |     f_goal_(boost::none), absolute_error_(sqrt(std::numeric_limits<double>::epsilon()))
      |                                                     ^~~~~~~~~~~~~~
minimizer.cc:29:68: error: expected primary-expression before 'double'
   29 |     f_goal_(boost::none), absolute_error_(sqrt(std::numeric_limits<double>::epsilon()))
      |                                                                    ^~~~~~
minimizer.cc: In member function 'Minimizer* Minimizer::set_absolute_error(double)':
minimizer.cc:93:65: error: 'numeric_limits' is not a member of 'std'
   93 |         absolute_error_ = std::max(an_absolute_error, sqrt(std::numeric_limits<double>::epsilon()));
      |                                                                 ^~~~~~~~~~~~~~
minimizer.cc:93:80: error: expected primary-expression before 'double'
   93 |         absolute_error_ = std::max(an_absolute_error, sqrt(std::numeric_limits<double>::epsilon()));
      |                                                                                ^~~~~~
minimizer.cc: In member function 'virtual double Minimizer::absolute_error() const':
minimizer.cc:128:59: error: 'numeric_limits' is not a member of 'std'
  128 |     return absolute_error_ ? *absolute_error_ : sqrt(std::numeric_limits<double>::epsilon());
      |                                                           ^~~~~~~~~~~~~~
minimizer.cc:128:74: error: expected primary-expression before 'double'
  128 |     return absolute_error_ ? *absolute_error_ : sqrt(std::numeric_limits<double>::epsilon());
      |                                                                          ^~~~~~
minimizer.cc: In constructor 'MinimizerMultiDimensionNoDerivative::MinimizerMultiDimensionNoDerivative(const gsl_multimin_function&, const array_type&, const array_type&)':
minimizer.cc:362:31: error: 'numeric_limits' is not a member of 'std'
  362 |     characteristic_size_(std::numeric_limits<double>::max())
      |                               ^~~~~~~~~~~~~~
minimizer.cc:362:46: error: expected primary-expression before 'double'
  362 |     characteristic_size_(std::numeric_limits<double>::max())
      |                                              ^~~~~~
minimizer.cc: In constructor 'MinimizerMultiDimensionNoDerivative::MinimizerMultiDimensionNoDerivative(const gsl_multimin_function&, const array_type&)':
minimizer.cc:382:31: error: 'numeric_limits' is not a member of 'std'
  382 |     characteristic_size_(std::numeric_limits<double>::max())
      |                               ^~~~~~~~~~~~~~
minimizer.cc:382:46: error: expected primary-expression before 'double'
  382 |     characteristic_size_(std::numeric_limits<double>::max())
      |                                              ^~~~~~
g++ -DHAVE_CONFIG_H -I. -I..    -DENFUSE_SOURCE   -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -I../src/dynamic_loader -I../src/layer_selection -g -O2 -O2 -DNDEBUG -c -o enfuse-error_message.o `test -f 'error_message.cc' || echo './'`error_message.cc
g++ -DHAVE_CONFIG_H -I. -I..    -DENFUSE_SOURCE   -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -I../src/dynamic_loader -I../src/layer_selection -g -O2 -O2 -DNDEBUG -c -o enfuse-filenameparse.o `test -f 'filenameparse.cc' || echo './'`filenameparse.cc
make[4]: *** [Makefile:827: enblend-minimizer.o] Error 1
make[4]: *** Waiting for unfinished jobs....
make[4]: Leaving directory '/build/enblend-enfuse-4.2/src'
make[3]: *** [Makefile:1096: all-recursive] Error 1
make[3]: Leaving directory '/build/enblend-enfuse-4.2/src'
make[2]: *** [Makefile:520: all] Error 2
make[2]: Leaving directory '/build/enblend-enfuse-4.2/src'
make[1]: *** [Makefile:432: all-recursive] Error 1
make[1]: Leaving directory '/build/enblend-enfuse-4.2'
make: *** [Makefile:372: all] Error 2
error: builder for '/nix/store/zlr7bmh11d6m8hjzfanvnm9jyvm5s0zl-enblend-enfuse-4.2.drv' failed with exit code 2;
```

Removing `enblend-enfuse` as a hugin and digikam dependency seems to build.

```
diff --git a/pkgs/applications/graphics/digikam/default.nix b/pkgs/applications/graphics/digikam/default.nix
index 141f966253d..645a0b00cb7 100644
--- a/pkgs/applications/graphics/digikam/default.nix
+++ b/pkgs/applications/graphics/digikam/default.nix
@@ -45,7 +45,6 @@
 , x265
 
 # For panorama and focus stacking
-, enblend-enfuse
 , hugin
 , gnumake
 
@@ -121,7 +120,7 @@ mkDerivation rec {
 
   preFixup = ''
     qtWrapperArgs+=("''${gappsWrapperArgs[@]}")
-    qtWrapperArgs+=(--prefix PATH : ${lib.makeBinPath [ gnumake hugin enblend-enfuse ]})
+    qtWrapperArgs+=(--prefix PATH : ${lib.makeBinPath [ gnumake hugin ]})
     qtWrapperArgs+=(--suffix DK_PLUGIN_PATH : ${placeholder "out"}/${qtbase.qtPluginPrefix}/${pname})
     substituteInPlace $out/bin/digitaglinktree \
       --replace "/usr/bin/perl" "${perl}/bin/perl" \
diff --git a/pkgs/applications/graphics/hugin/default.nix b/pkgs/applications/graphics/hugin/default.nix
index ec2ac709f86..437cd1e8a77 100644
--- a/pkgs/applications/graphics/hugin/default.nix
+++ b/pkgs/applications/graphics/hugin/default.nix
@@ -9,7 +9,6 @@
 , autopanosiftc
 , boost
 , cairo
-, enblend-enfuse
 , exiv2
 , fftw
 , flann
@@ -90,7 +89,6 @@ stdenv.mkDerivation rec {
     for p in $out/bin/*; do
       wrapProgram "$p" \
         --suffix PATH : ${autopanosiftc}/bin \
-        --suffix PATH : ${enblend-enfuse}/bin \
         --suffix PATH : ${gnumake}/bin \
         --suffix PATH : ${perlPackages.ImageExifTool}/bin
     done
```

### Notify maintainers

No direct maintainer, tagging @hrdinka since they are affected via hugin (`pkgs/applications/graphics/hugin/default.nix`); digikam has no current maintainer but is also affected.
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.6, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4`
 - channels(root): `"nixos-22.05pre336298.bc5d68306b4, nixos-small-22.05pre337082.499bd17576c"`
 - channels(fabian): `"home-manager, nixos-19.03-19.03.173691.34c7eb7545d, nixos-19.09-19.09.2522.75f4ba05c63, nixos-20.03-20.03.3330.1db42b7fe38, nixos-20.09-20.09.4407.1c1f5649bb9, nixos-21.05-21.05.4448.4a8751d694b, nixos-21.11-21.11.334139.1bd4bbd49be"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -3960,7 +3960,7 @@ with pkgs;
 
   libceph = ceph.lib;
   inherit (callPackages ../tools/filesystems/ceph {
-    boost = boost17x.override { enablePython = true; python = python3; };
+    boost = boost175.override { enablePython = true; python = python3; };
     lua = lua5_4;
   })
     ceph
@@ -5000,13 +5000,17 @@ with pkgs;
 
   emv = callPackage ../tools/misc/emv { };
 
-  enblend-enfuse = callPackage ../tools/graphics/enblend-enfuse { };
+  enblend-enfuse = callPackage ../tools/graphics/enblend-enfuse {
+    boost = boost172;
+  };
 
   endlessh = callPackage ../servers/endlessh { };
 
   ericw-tools = callPackage ../applications/misc/ericw-tools { };
 
-  cryfs = callPackage ../tools/filesystems/cryfs { };
+  cryfs = callPackage ../tools/filesystems/cryfs {
+    boost = boost175;
+  };
 
   encfs = callPackage ../tools/filesystems/encfs {
     tinyxml2 = tinyxml-2;
@@ -6578,7 +6582,9 @@ with pkgs;
   isl_0_20 = callPackage ../development/libraries/isl/0.20.0.nix { };
   isl_0_24 = callPackage ../development/libraries/isl/0.24.0.nix { };
 
-  ispike = callPackage ../development/libraries/science/robotics/ispike { };
+  ispike = callPackage ../development/libraries/science/robotics/ispike {
+    boost = boost16x;
+  };
 
   isrcsubmit = callPackage ../tools/audio/isrcsubmit { };
 
@@ -8735,7 +8741,9 @@ with pkgs;
 
   postscript-lexmark = callPackage ../misc/drivers/postscript-lexmark { };
 
-  povray = callPackage ../tools/graphics/povray { };
+  povray = callPackage ../tools/graphics/povray {
+    boost = boost175;
+  };
 
   power-profiles-daemon = callPackage ../os-specific/linux/power-profiles-daemon { };
 
@@ -13006,7 +13014,9 @@ with pkgs;
 
   smlpkg = callPackage ../tools/package-management/smlpkg { };
 
-  solc = callPackage ../development/compilers/solc { };
+  solc = callPackage ../development/compilers/solc {
+    boost = boost172;
+  };
 
   souffle = callPackage ../development/compilers/souffle {
     autoreconfHook = buildPackages.autoreconfHook269;
@@ -15418,7 +15428,9 @@ with pkgs;
   uefi-firmware-parser = callPackage ../development/tools/analysis/uefi-firmware-parser { };
 
   uhd3_5 = callPackage ../applications/radio/uhd/3.5.nix { };
-  uhd = callPackage ../applications/radio/uhd { };
+  uhd = callPackage ../applications/radio/uhd {
+    boost = boost175;
+  };
 
   uisp = callPackage ../development/embedded/uisp { };
 
@@ -15800,7 +15812,9 @@ with pkgs;
 
   captive-browser = callPackage ../applications/networking/browsers/captive-browser { };
 
-  ndn-cxx = callPackage ../development/libraries/ndn-cxx { };
+  ndn-cxx = callPackage ../development/libraries/ndn-cxx {
+    boost = boost175;
+  };
 
   ndn-tools = callPackage ../tools/networking/ndn-tools { };
 
@@ -17633,7 +17647,9 @@ with pkgs;
 
   libgtop = callPackage ../development/libraries/libgtop {};
 
-  libLAS = callPackage ../development/libraries/libLAS { };
+  libLAS = callPackage ../development/libraries/libLAS {
+    boost = boost172;
+  };
 
   liblaxjson = callPackage ../development/libraries/liblaxjson { };
 
@@ -17854,7 +17870,9 @@ with pkgs;
 
   librevisa = callPackage ../development/libraries/librevisa { };
 
-  librime = callPackage ../development/libraries/librime {};
+  librime = callPackage ../development/libraries/librime {
+    boost = boost174;
+  };
 
   librsb = callPackage ../development/libraries/librsb {
     # Taken from https://build.opensuse.org/package/view_file/science/librsb/librsb.spec
@@ -18046,7 +18064,9 @@ with pkgs;
 
   libndctl = callPackage ../development/libraries/libndctl { };
 
-  libnest2d = callPackage ../development/libraries/libnest2d { };
+  libnest2d = callPackage ../development/libraries/libnest2d {
+    boost = boost174;
+  };
 
   libnet = callPackage ../development/libraries/libnet { };
 
@@ -20109,7 +20129,9 @@ with pkgs;
   wlr-protocols = callPackage ../development/libraries/wlroots/protocols.nix { };
 
   wt = wt4;
-  inherit (callPackages ../development/libraries/wt {})
+  inherit (callPackages ../development/libraries/wt {
+    boost = boost175;
+  })
     wt3
     wt4;
 
@@ -21469,7 +21491,9 @@ with pkgs;
   };
 
   # Fails to compile with boost <= 1.72
-  rippled = callPackage ../servers/rippled { };
+  rippled = callPackage ../servers/rippled {
+    boost = boost172;
+  };
 
   rippled-validator-keys-tool = callPackage ../servers/rippled/validator-keys-tool.nix { };
 
@@ -27412,7 +27436,9 @@ with pkgs;
 
   openfx = callPackage ../development/libraries/openfx {};
 
-  openimageio = callPackage ../applications/graphics/openimageio { };
+  openimageio = callPackage ../applications/graphics/openimageio {
+    boost = boost175;
+  };
 
   openimageio2 = callPackage ../applications/graphics/openimageio/2.x.nix { };
 
@@ -28749,7 +28775,9 @@ with pkgs;
     swt = swt_jdk8;
   };
 
-  twister = callPackage ../applications/networking/p2p/twister { };
+  twister = callPackage ../applications/networking/p2p/twister {
+    boost = boost16x;
+  };
 
   twmn = libsForQt5.callPackage ../applications/misc/twmn { };
 
@@ -29620,7 +29648,9 @@ with pkgs;
 
   ### BLOCKCHAINS / CRYPTOCURRENCIES / WALLETS
 
-  aeon = callPackage ../applications/blockchains/aeon { };
+  aeon = callPackage ../applications/blockchains/aeon {
+    boost = boost172;
+  };
 
   alfis = callPackage ../applications/blockchains/alfis {
     inherit (darwin.apple_sdk.frameworks) Cocoa Security WebKit;
@@ -29709,8 +29739,14 @@ with pkgs;
   dcrd = callPackage ../applications/blockchains/dcrd { };
   dcrwallet = callPackage ../applications/blockchains/dcrwallet { };
 
-  digibyte = libsForQt514.callPackage ../applications/blockchains/digibyte { withGui = true; };
-  digibyted = callPackage ../applications/blockchains/digibyte { withGui = false; };
+  digibyte = libsForQt514.callPackage ../applications/blockchains/digibyte {
+    withGui = true;
+    boost = boost170;
+  };
+  digibyted = callPackage ../applications/blockchains/digibyte {
+    withGui = false;
+    boost = boost170;
+  };
 
   dogecoin  = callPackage ../applications/blockchains/dogecoin {
     boost = boost165;
@@ -29730,10 +29766,12 @@ with pkgs;
   elements  = libsForQt5.callPackage ../applications/blockchains/elements {
     miniupnpc = miniupnpc_2;
     withGui = true;
+    boost = boost175;
   };
   elementsd = callPackage ../applications/blockchains/elements {
     miniupnpc = miniupnpc_2;
     withGui = false;
+    boost = boost175;
   };
 
   ergo = callPackage ../applications/blockchains/ergo { };
@@ -29784,7 +29822,9 @@ with pkgs;
 
   masari = callPackage ../applications/blockchains/masari { boost = boost165; };
 
-  nano-wallet = libsForQt5.callPackage ../applications/blockchains/nano-wallet { };
+  nano-wallet = libsForQt5.callPackage ../applications/blockchains/nano-wallet {
+    boost = boost172;
+  };
 
   namecoin  = callPackage ../applications/blockchains/namecoin { withGui = true; };
   namecoind = callPackage ../applications/blockchains/namecoin { withGui = false; };
@@ -29823,7 +29863,9 @@ with pkgs;
 
   wasabibackend = callPackage ../applications/blockchains/wasabibackend { Nuget = dotnetPackages.Nuget; };
 
-  wownero = callPackage ../applications/blockchains/wownero { };
+  wownero = callPackage ../applications/blockchains/wownero {
+    boost = boost175;
+  };
 
   zcash = callPackage ../applications/blockchains/zcash { stdenv = llvmPackages_11.stdenv; };
 
@@ -33576,7 +33618,9 @@ with pkgs;
 
   NSPlist = callPackage ../development/libraries/NSPlist {};
 
-  PlistCpp = callPackage ../development/libraries/PlistCpp {};
+  PlistCpp = callPackage ../development/libraries/PlistCpp {
+    boost = boost172;
+  };
 
   xib2nib = callPackage ../development/tools/xib2nib {};
 

